src/main/java/com/example/shopproject/controller/AdminController.java

```diff
@@ -0,0 +1,76 @@
+package com.example.shopproject.controller;
+
+import com.example.shopproject.entity.Product;
+import com.example.shopproject.entity.Product_category;
+import com.example.shopproject.service.ProductService;
+import com.example.shopproject.service.Product_categoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+public class AdminController {
+
+    @Autowired
+    ProductService productService;
+
+    @Autowired
+    Product_categoryService product_categoryService;
+
+    @PostMapping("/product")
+    public Product createProduct(@Valid @RequestBody Product product) {
+        return productService.createProduct(product);
+    }
+
+    @GetMapping("/product/all")
+    public List<Product> getAllProducts(){
+        return productService.getAllProducts();
+    }
+
+    @GetMapping("/product/{productID}")
+    public Product getProductById(@PathVariable Long productID) {
+        return productService.getProductById(productID);
+    }
+
+    @PutMapping("/product")
+    public Product updateProductById(@Valid @RequestBody Product product) {
+        return productService.updateProdictById(product);
+    }
+
+    @DeleteMapping("/product/{productID}")
+    public ResponseEntity<String> deleteProductById(@PathVariable Long productID) {
+        productService.deleteProductById(productID);
+        return new ResponseEntity<>("Успешно удален", HttpStatus.OK);
```

Comment:
you dont need to use ResponseEntity, it can be just void, status will be indicator of successfully finished transaction